packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.js

```diff
@@ -127,6 +128,13 @@ export default ({ api, coreSagas }) => {
       yield call(coreSagas.data.xlm.fetchLedgerDetails)
       yield call(coreSagas.data.xlm.fetchData)
 
+      const countryCodeR = yield select(selectors.core.settings.getCountryCode)
+      const countryCode = countryCodeR.getOrElse('US')
+      const currency = getCurrencyBasedOnCountry(countryCode)
+      yield put(actions.preferences.setSBFiatCurrency(currency))
+      // TODO - update wallet with currency should be done on BE side
+      yield put(actions.core.settings.setCurrency(currency))
```

Comment:
I see this as techDebt reason is there we have to set it back to wallet because otherwise it will be overwritten with https://github.com/blockchain/blockchain-wallet-v4-frontend/blob/development/packages/blockchain-wallet-v4-frontend/src/data/preferences/sagas.ts#L24-L38

Reply:
@milan-bc I think we should only run this if `firstLogin` is true

Reply:
Can we just remove sbFiatCurrency and assume from now on that your sbFiatCurrency === walletCurrency?